### PR TITLE
implement Redux-saga 

### PR DESCRIPTION
**Pull request by arun-kumar-codes**

### CODE REVIEW

File: src/api/api.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import axios from 'axios';
+
+axios.defaults.baseURL = 'http://144.76.34.244:8090/';
```
Comment (Rahul-Khera-Codes) [inline src/api/api.js]: use this url from config

File: src/api/api.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+export const loginApi = action => axios({
```
Comment (Rahul-Khera-Codes) [inline src/api/api.js]: define generic function  which will accept method, url ,data

File: src/api/api.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  method: 'post',
+  url: 'app_get_candidate',
+  data: {
+    email_id: action.payload.email,
+  },
+});
+
+export const devieDataApi = action => axios({
```
Comment (Rahul-Khera-Codes) [inline src/api/api.js]: spelling  mistake

File: src/reducers/user.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    isSuccess: { $set: false },
+    errors: { $set: action.payload.data },
+    isError: { $set: true },
+  },
+});
+
+const userLogoutSuccess = (state, action) => update(state, {
+  login: {
+    data: { $set: action.payload.data },
+    isLoading: { $set: false },
```
Comment (arun-kumar-codes) [inline src/reducers/user.js]: this is not correct. 

need to create helpers using immutable using the $extend function

Comment (arun-kumar-codes) [reply src/reducers/user.js]: its very important to fix

**Pull request merged**